c_src/filewatch_inotify.c

```diff
@@ -88,27 +105,30 @@ static void stop(ErlDrvData self_)
 
 static ErlDrvSSizeT call(
     ErlDrvData self_,
-    unsigned int UNUSED,
+    unsigned int operation,
     char *buf, ErlDrvSizeT UNUSED,
     char **rbuf, ErlDrvSizeT rlen,
     unsigned int *UNUSED)
 {
+    if (operation != PORT_CALL_ARG) goto fail_op_check;
+
     struct instance *self = (struct instance *)self_;
 
     int index = 0;
-    if (ei_decode_version(buf, &index, NULL) < 0) return -1;
+    if (ei_decode_version(buf, &index, NULL) < 0) goto fail_decode;
 
     int type;
     int path_len;
-    if (ei_get_type(buf, &index, &type, &path_len) < 0) return -1;
+    if (ei_get_type(buf, &index, &type, &path_len) < 0) goto fail_decode;
 
     assert(path_len >= 0);
-    char path[path_len+1];
+
+    char *path = alloca(path_len+1);
```

Comment:
Why the change to `alloca`? Was that because the compiler was whining about something.